src/common/Motion/MotionPath.tsx

```diff
@@ -3,38 +3,37 @@ import { motion, useMotionValue, useSpring } from 'framer-motion';
 import { interpolate } from 'd3-interpolate';
 import { DEFAULT_TRANSITION } from './config';
 
-export const MotionPath = ({
-  custom,
-  transition,
-  onAnimationFinished,
-  ...rest
-}) => {
+export const MotionPath = ({ custom, transition, ...rest }) => {
   const d = useMotionValue(custom.exit.d);
-  const prevPathRef = useRef(custom.exit.d);
-  const spring = useSpring(useMotionValue(prevPathRef.current), {
-    ...DEFAULT_TRANSITION,
-    from: 0,
-    to: 1
+  const prevEnterPathRef = useRef(custom.enter.d);
+  const prevSpringRef = useRef(0);
+  const spring = useSpring(0, {
+    ...DEFAULT_TRANSITION
   });
 
   useEffect(() => {
-    const interpolator = interpolate(prevPathRef.current, custom.enter.d);
+    spring.set(1);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  useEffect(() => {
+    let interpolator = interpolate(custom.exit.d, custom.enter.d);
+
+    // if the new path to animate towards (custom.enter.d) updates
+    // then update the interpolator to start from where the path is now
+    if (prevEnterPathRef.current !== custom.enter.d) {
+      interpolator = interpolate(d.get(), custom.enter.d);
+      prevSpringRef.current = spring.get();
+      prevEnterPathRef.current = custom.enter.d;
+      spring.set(prevSpringRef.current + 1);
+    }
 
     const unsub = spring.onChange((v) => {
-      d.set(interpolator(v));
-
-      // NOTE: This is tricky logic ( Also refer to MotionBar.tsx ) ...
-      //  - Must animate in only once ( renders )
-      //  - Must not animate when other props change ( tooltip hover )
-      //  - Must animate from prev to new position on updates ( live updates )
-      if (v === 1) {
-        prevPathRef.current = custom.enter.d;
-        onAnimationFinished();
-      }
+      d.set(interpolator(v - prevSpringRef.current));
     });
 
     return unsub;
-  });
+  }, [custom.enter.d, custom.exit.d, d, spring]);
```

Comment:
This basic logic for this change is:
- Start spring value at 0 and set it to start animating towards 1 on mount
- Only If the point that the path is animating towards changes, bump spring + 1  to have it start animating again
    - Update the interpolator to start at the current path and end at the new destination path

This change also allowed me to revert my previous change for fixing chart animations that were getting interrupted/restarted on hover. Because spring doesn't reset on re-renders now, only when the destination point is updated